Cannot build samples/net/ipv4_autoconf

Building samples/net/ipv4_autoconf for sam_e70_explained fails with errors. Upstream is at 1eead93a483bbb758e8ad418f3f715b4fe6d285c.
```
$ cd samples/net/ipv4_autoconf
$ make build && cd build
$ cmake -GNinja -DBOARD=sam_e70_xplained ..
$ ninja
```
...
```
zephyr/drivers/ethernet/eth_sam_gmac.c:60:2: error: #error Remaining free RX data buffers (CONFIG_NET_BUF_RX_COUNT -
 #error Remaining free RX data buffers (CONFIG_NET_BUF_RX_COUNT -
  ^~~~~
In file included from <command-line>:
zephyr/samples/net/ipv4_autoconf/build/zephyr/include/generated/autoconf.h:138:42: error: expected identifier or '(' before numeric constant
 #define CONFIG_ETH_SAM_GMAC_BUF_RX_COUNT 12
                                          ^~
zephyr/drivers/ethernet/eth_sam_gmac.c:61:2: note: in expansion of macro 'CONFIG_ETH_SAM_GMAC_BUF_RX_COUNT'
  CONFIG_ETH_SAM_GMAC_BUF_RX_COUNT) * CONFIG_NET_BUF_DATA_SIZE
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
zephyr/drivers/ethernet/eth_sam_gmac.c:1982:12: error: 'rx_desc_que0' undeclared here (not in a function); did you mean 'tx_desc_que0'?
     .buf = rx_desc_que0,
            ^~~~~~~~~~~~
            tx_desc_que0
[68/143] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/exc_manage.c.obj
```

Still failing with:
```
zephyr/drivers/ethernet/eth_sam_gmac.c:60:2: error: #error (CONFIG_NET_BUF_RX_COUNT - CONFIG_ETH_SAM_GMAC_BUF_RX_COUNT) * CONFIG_NET_BUF_DATA_SIZE are not large enough to hold a full frame
 #error (CONFIG_NET_BUF_RX_COUNT - CONFIG_ETH_SAM_GMAC_BUF_RX_COUNT) * \
  ^~~~~
[58/150] Building C object zephyr/arch/arch/arm/core/CMakeFiles/arch__arm__core.dir/swap.c.obj
```

--- a/samples/net/ipv4_autoconf/boards/sam_e70_xplained.conf
+++ b/samples/net/ipv4_autoconf/boards/sam_e70_xplained.conf
@@ -1,0 +1,6 @@
+# sam_e70_xplained board need more buffers, otherwise it fails to build
+
+CONFIG_NET_PKT_RX_COUNT=20
+CONFIG_NET_PKT_TX_COUNT=20
+CONFIG_NET_BUF_RX_COUNT=30
+CONFIG_NET_BUF_TX_COUNT=30

